Annotations/.NETCore/System.Private.CoreLib/Attributes.xml

```diff
@@ -6072,6 +6072,24 @@
   <member name="M:System.Runtime.InteropServices.Marshal.PtrToStringUni(System.IntPtr)">
     <attribute ctor="M:JetBrains.Annotations.CanBeNullAttribute.#ctor"/>
   </member>
+  <member name ="M:System.Runtime.InteropServices.Marshal.PtrToStructure(System.IntPtr, System.Type)">
+    <attribute ctor="M:JetBrains.Annotations.ContractAnnotationAttribute.#ctor(System.String)">
+      <argument>ptr:null=&gt;null</argument>
```

Comment:
"ptr" is a struct, it cannot be null. There is no way to define contract like "IntPtr.Zero => null"

Reply:
It does work as expected.
Try this code with ReSharper:

```cs
using System;
using JetBrains.Annotations;

static class M
{
	[NotNull]
	[ContractAnnotation("p:null => null")]
	public static object F(IntPtr p)
	{
		// ReSharper disable once AssignNullToNotNullAttribute
		return p == IntPtr.Zero ? null : (object) 1;
	}
}

public static class Program
{
	static void Main(string[] args)
	{
		object a = M.F(IntPtr.Zero);
		var t = a.GetType(); // Possible NRE warning for 'a'

		object b = M.F((IntPtr)1);
		var w = b.GetType(); // No Possible NRE warning
	}
}
```

![image](https://user-images.githubusercontent.com/580536/53491683-ab520100-3a9f-11e9-8a30-42db3096e22f.png)

Reply:
this is just a coincidence while processing contradicting annotations "NotNull" and "ContractAnnotation("null => null")" . Following two samples does not have expected behavior:
` if (ptr != IntPtr.Zero)
    {
      object b = M.F(ptr);
      var w = b.GetType(); // Possible NRE warning
    }

object c = M.F((IntPtr)0);
    var w1 = c.GetType();` // No Possible NRE warning

Reply:
IMO ReSharper should handle this correctly.

Reply:
@SirDuke So the only fix is to use [CanBeNull] which makes a lot of code to get Possible NRE. Right ?

Reply:
There are two options: add "CanBeNull" annotation and show a lot of "possible NRE" or omit any nullable annotation and not warn users that the method can return null in case of IntPtr.Zero. Looks like second option is better. What do you think?

Reply:
I guess so. I think it depends how it is defined in C# 8. I guess it will be defined as not returning null either.

Reply:
@SirDuke btw what is the status of this project ?
It seems like it is not so maintained.

Reply:
@NN--- which one? "ExternalAnnotations" ? It is updated from time to time and new releases are published

Reply:
Yeah. There are several issues here staying untesolved for much time.